broadway/api/handlers/client.py

```diff
@@ -270,3 +270,27 @@ def get(self, *args, **kwargs):
                 {"message": "scope {} has not been implemented yet".format(scope)}, 404
             )
             return
+
+
+class CourseQueueLengthHandler(ClientAPIHandler):
+    @authenticate_course
+    @schema.validate(
+        output_schema={
+            "type": "object",
+            "properties": {"length": {"type": "number"}},
+            "required": ["length"],
+            "additionalProperties": False,
+        },
+        on_empty_404=True,
+    )
+    def get(self, *args, **kwargs):
+        course_id = kwargs["course_id"]
+        queue = self.settings["QUEUE"]
+
+        if queue.contains_key(course_id):
+            return {"length": queue.get_queue_length_by_key(course_id)}
+        else:
+            self.abort(
+                {"message": f"{course_id} does not exist as a course in the queue"}
+            )
+            return
```

Comment:
Based off the previous point I made, it might be good to first check if the `course_id` exists in **configs**. If no such course exists, then return an error. If a course exists but queue doesn't contain its key, then return length 0.